lib/member_name_parts.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+require 'scraped'
+
+class MemberNameParts < Scraped::HTML
+  field :prefix do
+    partitioned.first
+  end
+
+  field :name do
+    # .sub called for specific case where Dr. is not separated from name by space
+    partitioned.last.sub('Dr.Bharrat', 'Bharrat')
+  end
+
+  field :gender do
+    prefixes_chomped.map { |p| gender_map[p] }.compact.first
+  end
+
+  private
+
+  def prefixes
+    %w(Assoc Prof Professor Rev Bishop Prince Dr Lt Col Colonel (Ret’)).to_set
+                                                                       .merge(female_prefixes)
+                                                                       .merge(male_prefixes)
+  end
+
+  def female_prefixes
+    %w(Mrs Ms Miss)
+  end
+
+  def male_prefixes
+    %w(Mr)
+  end
+
+  def gender_map
+    Hash[female_prefixes.map { |e| [e, 'female'] }].merge(Hash[male_prefixes.map { |e| [e, 'male'] }])
```

Comment:
As this is now using a more modern Ruby, you should be able to simplify these with `.to_h`